jerry-core/ecma/builtin-objects/ecma-builtin-function-prototype.cpp

```diff
@@ -226,7 +226,151 @@ ecma_builtin_function_prototype_object_bind (ecma_value_t this_arg, /**< this ar
+
+  /* 2. */
+  if (!ecma_op_is_callable (this_arg))
+  {
+    ret_value = ecma_make_throw_obj_completion_value (ecma_new_standard_error (ECMA_ERROR_TYPE));
+  }
+  else
+  {
+    /* 4. 11. 18. */
+    ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_FUNCTION_PROTOTYPE);
+    ecma_object_t *function_p = ecma_create_object (prototype_obj_p, true, ECMA_OBJECT_TYPE_BOUND_FUNCTION);
+
+    ecma_deref_object (prototype_obj_p);
+
+    /* 7. */
+    ecma_property_t *target_function_prop_p;
+    target_function_prop_p = ecma_create_internal_property (function_p,
+                                                            ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_TARGET_FUNCTION);
+
+    target_function_prop_p->u.internal_property.value = ecma_copy_value (this_arg, false);
+
+    /* 8. */
+    ecma_property_t *bound_this_prop_p;
+    bound_this_prop_p = ecma_create_internal_property (function_p, ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS);
+
+    size_t arg_list_size;
+
+    if (arguments_number > 0)
+    {
+      bound_this_prop_p->u.internal_property.value = ecma_copy_value (arguments_list_p[0], false);
+
+      arg_list_size = (size_t) (arguments_number) * sizeof (ecma_value_t);
+    }
+    else
+    {
+      ecma_object_t *glob_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_GLOBAL);
+      bound_this_prop_p->u.internal_property.value = ecma_copy_value (ecma_make_object_value (glob_obj_p), false);
+      ecma_deref_object (glob_obj_p);
+
+      arg_list_size = (size_t) sizeof (ecma_value_t);
+    }
+
+    /* 9. */
+    ecma_value_t *arg_list = static_cast <ecma_value_t *> (mem_heap_alloc_block (arg_list_size,
+                                                                                 MEM_HEAP_ALLOC_SHORT_TERM));
+
+    ecma_number_t *count = ecma_alloc_number ();
+    *count = ecma_uint32_to_number (arguments_number);
+
+    if (arguments_number > 0)
+    {
+      (*count)--;
+    }
+
+    /*
+     * Set the first element of the arguments array to the count of optional arguments.
+     * We need the number of bounded arguments, when the target function is called.
+     */
+    arg_list[0] = ecma_make_number_value (count);
+
+    for (uint32_t i = 1; i < arguments_number; i++)
+    {
+      arg_list[i] = ecma_copy_value (arguments_list_p[i], false);
+    }
+
+    ecma_property_t *bound_args_prop_p;
+    bound_args_prop_p = ecma_create_internal_property (function_p, ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_ARGS);
+    ECMA_SET_POINTER (bound_args_prop_p->u.internal_property.value, arg_list);
+
+    /*
+     * [[Class]] property is not stored explicitly for objects of ECMA_OBJECT_TYPE_FUNCTION type.
+     *
+     * See also: ecma_object_get_class_name
+     */
+
+    /* 17. */
+    ecma_number_t *length_p = ecma_alloc_number ();
+    *length_p = 0;
+
+    ecma_property_descriptor_t length_prop_desc = ecma_make_empty_property_descriptor ();
+    length_prop_desc.is_value_defined = true;
```

Comment:
Could you, please, add `{` `}` block for the property descriptor initialization code, like for `prop_desc` below?